File: app/src/main/java/com/artmurka/artmurkaapp/Views/Fragments/CategorySettings.java
package com.artmurka.artmurkaapp.Views.Fragments;


import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.RadioGroup;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import com.artmurka.artmurkaapp.Model.Databases.Preferences;
import com.artmurka.artmurkaapp.Presenter.CategorySettingsPresenter;
import com.artmurka.artmurkaapp.Presenter.InterfacesPresenter.ICategorySettingsPresenter;
import com.artmurka.artmurkaapp.R;
import com.artmurka.artmurkaapp.Views.Fragments.Interfaces.ICategorySettings;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;


/**
 * A simple {@link Fragment} subclass.
 */
public class CategorySettings extends Fragment implements ICategorySettings {
    private ICategorySettingsPresenter presenter;
    private ImageView ivMicrosoft, ivList;
    private Button btnApply;
    private RadioGroup radioGroup;

    private Spinner spinnerDate, spinnerPrice;
    private Integer currentSelect = 2;

    public CategorySettings() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        presenter = new CategorySettingsPresenter(this);
        View view = inflater.inflate(R.layout.fragment_category_settings, container, false);

        btnApply = (Button) view.findViewById(R.id.btnApply);
        ivList = (ImageView) view.findViewById(R.id.ivList);
        ivMicrosoft = (ImageView) view.findViewById(R.id.ivMicrosoft);
        radioGroup = (RadioGroup) view.findViewById(R.id.radioGroup);
        radioGroup.check(Preferences.getCheckedRadioButton());
        setIvListener();
        changeIconsColor(Preferences.getListSettings());
        btnApply.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                HashMap<String, String> settings = new HashMap<String, String>();
                Preferences.setCheckedRadioButton(radioGroup.getCheckedRadioButtonId());
                switch (radioGroup.getCheckedRadioButtonId()) {
                    case R.id.rbLowCost:
                        Toast.makeText(view.getContext(), "Дешево!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "price");
                        settings.put("order", "ask");
                        break;
                    case R.id.rbExpensive:
                        Toast.makeText(view.getContext(), "дорого!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "price");
                        settings.put("order", "desc");
                        break;
                    case R.id.rbName:
                        Toast.makeText(view.getContext(), "По имени!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "name");
                        settings.put("order", "ask");
                        break;
                    case R.id.rvNew:
                        Toast.makeText(view.getContext(), "Новые!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "added_time");
                        settings.put("order", "ask");
                        break;
                    case R.id.rbOld:
                        Toast.makeText(view.getContext(), "Старые!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "added_time");
                        settings.put("order", "desc");
                        break;
                    default:
                        Toast.makeText(view.getContext(), "По имени!", Toast.LENGTH_SHORT).show();
                        settings.put("sort", "name");
                        settings.put("order", "ask");
                        break;
                }
                presenter.applyChanges(settings, currentSelect);
                Preferences.setListSettings(currentSelect);
                FragmentManager fm = getActivity().getSupportFragmentManager();
                ItemListFragment itemList = new ItemListFragment();
                Bundle bundle = new Bundle();
                bundle.putString("url", Preferences.getListUrl());
                bundle.putString("sort", settings.get("sort"));
                bundle.putString("order", settings.get("order"));
                bundle.putInt("list", currentSelect);
                itemList.setArguments(bundle);
                fm.beginTransaction()
                        .replace(R.id.mainFrame, itemList)
                        .addToBackStack(null)
                        .commit();
                fm.executePendingTransactions();
            }
        });
        return view;
    }

    private void setIvListener() {

        ivList.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                changeIconsColor(1);
            }
        });

        ivMicrosoft.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                changeIconsColor(2);
            }
        });

    }

    @Override
    public void onResume() {
        super.onResume();
        try {
            ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle("Фільтри");
        } catch (NullPointerException e) {
            e.printStackTrace();
        }
    }


    private void changeIconsColor(Integer id) {

        this.currentSelect = id;

        if (id == 1) {
            ivList.setImageDrawable(getResources().getDrawable(R.drawable.microsoft_pink));
            ivMicrosoft.setImageDrawable(getResources().getDrawable(R.drawable.view_list));
        } else {
            ivList.setImageDrawable(getResources().getDrawable(R.drawable.microsoft));
            ivMicrosoft.setImageDrawable(getResources().getDrawable(R.drawable.viewlist_pink));
        }
    }
}

File: app/src/main/java/com/artmurka/artmurkaapp/android/views/fragments/CategorySettings.kt
package com.artmurka.artmurkaapp.android.views.fragments


import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.RadioGroup
import android.widget.Spinner
import android.widget.Toast

import com.artmurka.artmurkaapp.data.model.databases.Preferences
import com.artmurka.artmurkaapp.presenter.CategorySettingsPresenter
import com.artmurka.artmurkaapp.presenter.Presenter
import com.artmurka.artmurkaapp.presenter.PresenterView
import com.artmurka.artmurkaapp.R
import com.artmurka.artmurkaapp.android.views.fragments.interfaces.ICategorySettings
import kotlinx.android.synthetic.main.fragment_category_settings.*

import java.util.HashMap
import javax.inject.Inject

class CategorySettings : BaseFragment(), ICategorySettings {

    @Inject
    lateinit var presenter: CategorySettingsPresenter


    private var currentSelect: Int? = 2

    override fun getLayout(): Int {
        return R.layout.fragment_category_settings
    }

    override fun getFragmentPresenter(): Presenter<out PresenterView> = presenter

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        presenter.takeView(this)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        radioGroup.check(Preferences.checkedRadioButton!!)
        setIvListener()
        changeIconsColor(Preferences.listSettings!!)
        btnApply.setOnClickListener { view ->
            val settings = HashMap<String, String>()
            Preferences.checkedRadioButton = radioGroup!!.checkedRadioButtonId
            when (radioGroup!!.checkedRadioButtonId) {
                R.id.rbLowCost -> {
                    Toast.makeText(view.context, resources.getString(R.string.fragment_category_settings_cheap), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "price"
                    settings["order"] = "ask"
                }
                R.id.rbExpensive -> {
                    Toast.makeText(view.context,  resources.getString(R.string.fragment_category_settings_expensive), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "price"
                    settings["order"] = "desc"
                }
                R.id.rbName -> {
                    Toast.makeText(view.context, resources.getString(R.string.fragment_category_settings_byname), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "name"
                    settings["order"] = "ask"
                }
                R.id.rvNew -> {
                    Toast.makeText(view.context, resources.getString(R.string.fragment_category_settings_new), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "added_time"
                    settings["order"] = "ask"
                }
                R.id.rbOld -> {
                    Toast.makeText(view.context, resources.getString(R.string.fragment_category_settings_old), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "added_time"
                    settings["order"] = "desc"
                }
                else -> {
                    Toast.makeText(view.context, resources.getString(R.string.fragment_category_settings_byname), Toast.LENGTH_SHORT).show()
                    settings["sort"] = "name"
                    settings["order"] = "ask"
                }
            }
            presenter!!.applyChanges(settings, currentSelect!!)
            Preferences.listSettings = currentSelect
            val fm = activity!!.supportFragmentManager
            val itemList = ItemListFragment()
            val bundle = Bundle()
            bundle.putString("url", Preferences.listUrl)
            bundle.putString("sort", settings["sort"])
            bundle.putString("order", settings["order"])
            bundle.putInt("list", currentSelect!!)
            itemList.arguments = bundle
            fm.beginTransaction()
                    .replace(R.id.mainFrame, itemList)
                    .addToBackStack(null)
                    .commit()
            fm.executePendingTransactions()
        }
    }


    private fun setIvListener() {

        ivList.setOnClickListener { changeIconsColor(1) }

        ivMicrosoft.setOnClickListener { changeIconsColor(2) }
        ivGrid.setOnClickListener { changeIconsColor(3) }

    }

    override fun onResume() {
        super.onResume()
        try {
            (activity as AppCompatActivity).supportActionBar!!.title = resources.getString(R.string.fragment_filters)
        } catch (e: NullPointerException) {
            e.printStackTrace()
        }

    }


    private fun changeIconsColor(id: Int) {

        this.currentSelect = id

        when (id) {
            1 -> {
                ivList!!.setImageDrawable(resources.getDrawable(R.drawable.microsoft_pink))
                ivMicrosoft!!.setImageDrawable(resources.getDrawable(R.drawable.view_list))
                ivGrid!!.setImageDrawable(resources.getDrawable(R.drawable.ic_grid_grey))
            }

            2 -> {
                ivList!!.setImageDrawable(resources.getDrawable(R.drawable.microsoft))
                ivMicrosoft!!.setImageDrawable(resources.getDrawable(R.drawable.viewlist_pink))
                ivGrid!!.setImageDrawable(resources.getDrawable(R.drawable.ic_grid_grey))
            }

            3 -> {
                ivList!!.setImageDrawable(resources.getDrawable(R.drawable.microsoft))
                ivMicrosoft!!.setImageDrawable(resources.getDrawable(R.drawable.view_list))
                ivGrid!!.setImageDrawable(resources.getDrawable(R.drawable.ic_grid_pink))
            }
        }
    }

}// Required empty public constructor
